### PR TITLE
[BACKLOG-36556] Fix for possible NPE in .equals implementation

### PR DESCRIPTION
**Pull request by urena-kenny**

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/browse
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

File: kettle-plugins/browse/src/main/java/org/pentaho/big/data/impl/browse/model/NamedClusterFile.java [hunk 1]
```diff
@@ -71,8 +71,10 @@ public static NamedClusterFile create( String parent, FileObject fileObject ) {
     }
 
     NamedClusterFile compare = (NamedClusterFile) obj;
+    // This comparison depends on `getProvider()` to always return a hardcoded value
     return compare.getProvider().equals( getProvider() )
-      && ( ( compare.getPath() == null && getPath() == null ) || compare.getPath().equals( getPath() ) );
+      && ( ( compare.getPath() == null && getPath() == null ) || ( compare.getPath() != null && compare.getPath()
```
Comment (NJtwentyone) [inline kettle-plugins/browse/src/main/java/org/pentaho/big/data/impl/browse/model/NamedClusterFile.java:76]: No change, could also use apache's StringUtils library to replace line 75
`StringUtils.equals(compare.getPath(), getPath())`

**Review by NJtwentyone — APPROVED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 2s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/browse
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 1s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/browse
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**